components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -507,14 +507,19 @@ private OAuth2IntrospectionResponseDTO validateAccessToken(OAuth2TokenValidation
                 }
             }
 
+            String tokenType = accessTokenDO.getTokenType();
+
             // should be in seconds
             introResp.setIat(accessTokenDO.getIssuedTime().getTime() / 1000);
             // Not before time will be the same as issued time.
             introResp.setNbf(accessTokenDO.getIssuedTime().getTime() / 1000);
             // token scopes
             introResp.setScope(OAuth2Util.buildScopeString((accessTokenDO.getScope())));
             // set user-name
-            introResp.setUsername(getAuthzUser(accessTokenDO));
+            if (!OAuthServerConfiguration.getInstance().isRemoveUsernameFromIntrospectionResponseForAppTokensEnabled()
```

Comment:
Would be good to assign them to variables and keep the condition clear.

An example suggestion,

```
boolean removeUsernameEnabled = OAuthServerConfiguration.getInstance().isRemoveUsernameFromIntrospectionResponseForAppTokensEnabled();
boolean isAppTokenType = StringUtils.equals(OAuthConstants.UserType.APPLICATION, tokenType);

if (!removeUsernameEnabled || !isAppTokenType) {
    introResp.setUsername(getAuthzUser(accessTokenDO));
}

```

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2088/commits/eac830d53d9f00ebefa78944532517deaeee7354